src/core/agent/services/identifierService.ts

```diff
@@ -303,6 +313,150 @@ class IdentifierService extends AgentService {
       );
     }
   }
+
+  async createMultisig(
+    creatorIdentifier: string,
+    contacts: KeriContact[],
+    meta: Pick<
+      IdentifierMetadataRecordProps,
+      "displayName" | "colors" | "theme"
+    >
+  ): Promise<string | undefined> {
+    const creatorMetadata = await this.getMetadataById(creatorIdentifier);
+    this.validIdentifierMetadata(creatorMetadata);
+    const creatorAid = (await this.agent.modules.signify.getIdentifierByName(
+      creatorMetadata.signifyName as string
+    )) as Aid;
+    const otherAids = [];
+    for (const contact of contacts) {
+      const aid = await this.agent.modules.signify.resolveOobi(contact.oobi);
+      otherAids.push({
+        state: aid.response,
+      } as Pick<Aid, "state">);
+    }
+    const result = await this.agent.modules.signify.createMultisig(
+      creatorAid,
+      otherAids
+    );
+    const multisigId = result.op.name.split(".")[1];
+    await this.createIdentifierMetadataRecord({
+      id: multisigId,
+      displayName: meta.displayName,
+      method: IdentifierType.KERI,
+      colors: meta.colors,
+      theme: meta.theme,
+      signifyName: utils.uuid(),
+      opName: result.op.name, //we save the opName here to sync the multisig's status later
+      isPending: result.op.done ? false : true, //this will be updated once the operation is done
+    });
+    return multisigId;
+  }
+  async hasJoinedMultisig(msgSaid: string): Promise<boolean> {
+    const notifications: MultiSigIcpNotification[] =
+      await this.agent.modules.signify.getNotificationsBySaid(msgSaid);
+    if (!notifications.length) {
+      return false;
+    }
+    const exn = notifications[0].exn;
+    const signifyName = exn.a.name;
```

Comment:
We set `signifyName` as a UUID when creating or joining, is this correct?

Reply:
Yes

Reply:
So `exn.a.name` is always the UUID we created? So the person who initiates the multi-sig decides the name, and everyone else sets it this same `signifyName`?

Reply:
After rethinking, I think the UUID must be the id of the multisig instead of the name that the creator created. Other users can choose another name for the multisig in their wallet.

Reply:
OK, so if we create it, we set a UUID as the name, but all other parties (if they use our wallet) will set it to the AID of the multi-sig?

Reply:
I've changed the UUID to the id of the multisig

Reply:
I think the AID of the multisig for the `signifyName` makes sense to me. This name is only used for fetching from KERIA so it's not user facing, so I think we can leave it as it is. Agree?

Reply:
I've changed the code to this. I think using this id would be better than the name.
```
const exn = notifications[0].exn;
const multisigId = exn.a.gid;
```

Reply:
Agree, but there's something wrong here.

Firstly:
- `createMultisig` for when we initiate: this calls `signifyApi.ts` and stores `signifyName: utils.uuid()` in the metadata record
  - `createMultisig` in `signifyApi.ts` creates the identifier on KERIA using `signifyName: utils.uuid();` - this is a new UUID, different to the one stored in the metadata record?

Here -> we should pass the UUID across.

Secondly:
- `joinMultisig` for joining a multi-sig initiated by someone else: this calls `signifyApi.ts` and stores `signifyName: utils.uuid()` in the metadata record
  - `joinMultisig` in `signifyApi.ts` creates the identifier on KERIA using `const name = exn.a.name;` not the UUID and not `exn.a.gid`

Thirdly:
- `joinMultisig` calls `hasJoinedMultisig` and checks for `exn.a.gid` not the UUID or `exn.a.gid`

Here, they should all use `enx.a.gid` and it should be passed back to the metadata record.

Reply:
Ah, I got your point. I will make an update for this.

Reply:
The `gid` in the third bullet point will be the same as the id in the metadata.

Reply:
So `gid` isn't the ID of the identifier, it's the SignifyName set by the person who creates it?

Still need to update some of these anyway as there are different UUIDs being generated for the same thing.

Reply:
No. I mean the gid is also the id of the identifier, the signifyName will be the UUID created by us. I have pushed the latest updates.

Reply:
OK - so now `createMultisig` and `joinMultisig` both use UUIDs and pass the UUID to `signifyApi.ts`.

But now I see - I didn't realise `hasJoinedMultisig` was fetching by ID and not fetching by Signify name. Makes sense now - KERIA really need an endpoint to fetch by ID instead of us fetching them all and searching the array in `findIdentifierById` - but anyway, that's a later problem. Good for now :)